### PR TITLE
Release candidate 3: HyperionSequencedReader & Memory leak fixes

### LINKED ISSUE #13: Fix memory leak on 1.5 rc2
One of the things holding up the release of EVM 1.5 is a memory leak we haven't been able to find the source of.

Current work is focused on changing the old code we inherited on the state history consumer layer from the repo `pinknetwork/eosio-contract-api`, to the new `hyperion-sequenced-reader` that Igor is working on, we hope that solves the issue as the old code did a lot of work in parallel threads using old non standard apis from node-worker-process package.

### PR DESCRIPTION
**Pull request by guilledk**
Fixes #13 #14

## Description

Swap `StateHistoryReader` from `pinknetwork/eosio-contract-api` to use new `HyperionSequencedReader` from `eosrio/hyperion-sequenced-reader` which is under development, we are hoping this helps to address the memory leak we encountered.

## Test scenarios

The indexer should be able to sync from the first block of the evm chain to head without errors or crashes of any kind and not leaking any memory.

## Checklist:

-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have cleaned up the code in the areas my change touches
-   [x] My changes generate no new warnings
-   [x] Any dependent changes have been merged and published in downstream modules
-   [x] I have checked my code and corrected any misspellings
-   [x] I have removed any unnecessary console messages

### CODE REVIEW

File: src/indexer.ts [hunk 1]
```diff
@@ -238,106 +227,202 @@ export class TEVMIndexer {
...
+                    this.rpc,
+                    gasUsedBlock
+                );
+            } else
+                continue;
+
+            if (!isTxDeserializationError(evmTx))
+                gasUsedBlock.iadd(new BN(evmTx.gasused, 10));
+
+            if (isTxDeserializationError(evmTx)) {
```
Comment (guilledk) [inline src/indexer.ts:367]: Always throw, check if debug mode is still necesary

File: src/indexer.ts [hunk 2]
```diff
@@ -580,39 +666,26 @@ export class TEVMIndexer {
      * Detect forks and handle them, leave every state tracking attribute in a healthy state
      */
     private async maybeHandleFork(b: ProcessedBlock) {
-        if (this.forked || b.nativeBlockNumber > this.lastNativeOrderedBlock)
+        if (this.forked ||
```
Comment (guilledk) [inline src/indexer.ts:669]: State handling flag not needed `forked`

**Review by guilledk — COMMENTED** [review]
co-review by @poplexity and me.

**Review by poplexity — APPROVED** [review]
👍

**Pull request merged**